kloppy/io.py

```diff
@@ -1,25 +1,79 @@
+"""I/O utilities for reading raw data."""
 import contextlib
 import logging
 import os
 import urllib.parse
 from dataclasses import dataclass, replace
+from io import BytesIO, BufferedIOBase
 from pathlib import PurePath
-from typing import Union, IO, BinaryIO, Tuple
-
-from io import BytesIO
+from typing import (
+    IO,
+    BinaryIO,
+    Tuple,
+    Union,
+    Optional,
+    Generator,
+    ContextManager,
+)
 
 from kloppy.config import get_config
 from kloppy.exceptions import InputNotFoundError
 from kloppy.infra.io.adapters import get_adapter
 
-
 logger = logging.getLogger(__name__)
 
-_open = open
+
+def _open(file: str, mode: str) -> IO:
+    if file.endswith(".gz"):
+        import gzip
+
+        return gzip.open(file, mode)
+    elif file.endswith(".xz"):
+        import lzma
+
+        return lzma.open(file, mode)
+    elif file.endswith(".bz2"):
+        import bz2
+
+        return bz2.open(file, mode)
+    return open(file, mode)
+
+
+def _decompress(stream: BinaryIO) -> BinaryIO:
+    stream.seek(0)
+    if stream.read(3) == b"\x1f\x8b\x08":
```

Comment:
Haven't tested this yet, but this should not work, right?

The first `read(3)` will change the file pointer, which will cause the next `read(6)` to start at offset 3.

Reply:
Indeed, good catch! It has been fixed now.